specification/network/resource-manager/Microsoft.Network/2017-09-01/applicationSecurityGroup.json

```diff
@@ -0,0 +1,370 @@
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "applicationSecurityGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the application security group."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationSecurityGroup"
+            },
+            "description": "Parameters supplied to the create or update ApplicationSecurityGroup operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Create successful. The operation returns the resulting application security group resource.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationSecurityGroup"
+            }
+          },
+          "200": {
+            "description": "Update successful. The operation returns the resulting application security group resource.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationSecurityGroup"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create application security group": { "$ref": "./examples/ApplicationSecurityGroupCreate.json" }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/applicationSecurityGroups": {
+      "get": {
+        "tags": [
+          "ApplicationSecurityGroups"
+        ],
+        "operationId": "ApplicationSecurityGroups_ListAll",
+        "description": "Gets all application security groups in a subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of application security group resources.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationSecurityGroupListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List all application security groups": { "$ref": "./examples/ApplicationSecurityGroupListAll.json" }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationSecurityGroups/{applicationSecurityGroupName}/ipConfigurations/$ref": {
```

Comment:
I am not sure if this $ref will work. We will have to confirm this early next week after the break.

Reply:
I thought the plan was to have this resolved today? Can we try and stick to that, please? I think the deadline for this to be checked in (for Ignite) is early next week, right?

Reply:
$ref will not work. Can we make it GET /applicationSecurityGroups/{applicationSecurityGroupName}/ipConfigurations and return a value array which has name and type properties in addition to id.